### PR TITLE
Create kops jobs for k8s 1.5 and 1.6

### PR DESCRIPTION
**Pull request by justinsb**
I still need to figure out how to link in the actual version (i.e. how to honor JENKINS_PUBLISHED_VERSION)

cc @zmerlynn

### CODE REVIEW

**Comment by ixdy** [comment]
x-ref #2092 #2029

**Comment by chrislovecnm** [comment]
@k8s-bot bazel test this

File: jobs/ci-kubernetes-e2e-kops-aws-release-1.5.env [hunk 1]
```diff
@@ -0,0 +1 @@
+KOPS_DEPLOY_LATEST_URL=https://storage.googleapis.com/kubernetes-release-dev/ci/latest-1.5.txt
```
Comment (krzyzacy) [inline jobs/ci-kubernetes-e2e-kops-aws-release-1.5.env]: hummm... unit-test asks for a KUBEKINS_TIMEOUT here, or you can make it a flag

File: jobs/config.json [hunk 2]
```diff
@@ -356,6 +356,26 @@
   ]
 },
 
+"ci-kubernetes-e2e-kops-aws-release-1.5": {
+  "scenario": "kubernetes_kops_aws",
+  "args": [
+    "--cluster=e2e-kops-aws-16.test-aws.k8s.io",
```
Comment (krzyzacy) [inline jobs/config.json]: 15?

Comment (justinsb) [inline jobs/config.json]: Thank you :-)  Who knows how long I would have wasted on that one!

File: jobs/ci-kubernetes-e2e-kops-aws-release-1.5.env (hunk 1 above)
Comment (justinsb) [inline jobs/ci-kubernetes-e2e-kops-aws-release-1.5.env]: My theory was that I would layer these, by passing multiple env args.  So I would do:
```
    "--env-file=platforms/kops_aws.env",
    "--env-file=jobs/ci-kubernetes-e2e-kops-aws.env",
    "--env-file=jobs/ci-kubernetes-e2e-kops-aws-release-1.5.env"
```

I'm not sure if that works or is a good idea....

File: jobs/config.json (hunk 2 above)
Comment (justinsb) [inline jobs/config.json]: Thank you :-) Who knows how long I would have wasted on that one!

File: jobs/ci-kubernetes-e2e-kops-aws-release-1.5.env (hunk 1 above)
Comment (krzyzacy) [inline jobs/ci-kubernetes-e2e-kops-aws-release-1.5.env]: yeah this works fine

**Comment by krzyzacy** [comment]
hummm assume you still need to solve `JENKINS_PUBLISHED_VERSION`?

**Comment by krzyzacy** [comment]
`--extract=ci/latest-1.6` flag in kubetest might work now

**Comment by justinsb** [comment]
Thanks @krzyzacy - I think I solved it by specifying e.g. `KOPS_DEPLOY_LATEST_URL=https://storage.googleapis.com/kubernetes-release-dev/ci/latest-1.6.txt`, but I'll have a look at the `extract` method as well.

But I also wanted to go over this with @zmerlynn first, just in case there's some AWS nuances I'm missing!

File: jobs/ci-kubernetes-e2e-kops-aws-release-1.5.env (hunk 1 above)
Comment (zmerlynn) [inline jobs/ci-kubernetes-e2e-kops-aws-release-1.5.env]: I love this layering, BTW.

**Comment by zmerlynn** [comment]
Presumably take the WIP out of the title. :)

**Comment by zmerlynn** [comment]
Hm. https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-kops-aws-release-1.6/1/build-log.txt - It looks like we may need both `KOPS_DEPLOY_LATEST_URL` and `JENKINS_PUBLISHED_VERSION`. The former influences what version `kops` deploys, but the latter is driving what version of tests we use.

**Comment by zmerlynn** [comment]
Verified. Looks like we're actually using the 1.6 tests now. (Presumably the 1.5 job will work, too, but I'm not going to bother kicking it right now.)

**Pull request merged**